src/services/createRoute.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { expect, test } from 'vitest'
+import { createRoute } from '@/services/createRoute'
+import { path } from '@/services/path'
+import { query } from '@/services/query'
+
+test('given parent, key is combined', () => {
+  const parent = createRoute({
+    name: 'parent',
+  })
+
+  const child = createRoute({
+    parent: parent,
+    name: 'child',
+  })
+
+  expect(child.key).toBe('parent.child')
+})
+
+test('given parent, path is combined', () => {
+  const parent = createRoute({
+    path: '/parent',
+  })
+
+  const child = createRoute({
+    parent: parent,
+    path: path('/child/[id]', { id: Number }),
+  })
+
+  expect(child.path).toMatchObject({
+    path: '/parent/child/[id]',
+    params: {
+      id: Number,
+    },
+  })
+})
+
+test('given parent, query is combined', () => {
+  const parent = createRoute({
+    query: 'static=123',
+  })
+
+  const child = createRoute({
+    parent: parent,
+    query: query('sort=[sort]', { sort: Boolean }),
+  })
+
+  expect(child.query).toMatchObject({
+    query: 'static=123&sort=[sort]',
+    params: {
+      sort: Boolean,
+    },
+  })
+})
+
+test('given parent, state is combined into stateParams', () => {
```

Comment:
every time I read `stateParams` I wonder why its not just `state` like `path` and `query` and `host`.

Reply:
well the issue stems from `ResolveRoute` needing to know both what the schema (params) for the state is, as well as having the ACTUAL runtime values for the state. This is unlike path/query/etc where once we go from Route -> Resolved the router no longer needs to know what the schema (params) is and if it did, it could use the `route.matched`. We might be able to store state on `route.matched`, but I think that would require a generic which I know you were opposed to.

honestly though, I prefer this delineation between the params and the runtime values. I would like to see us update `path` -> `pathParams`, `query` -> `queryParams`, etc

Reply:
resolved sync